### PR TITLE
Adding cache capability to avoid network calls through annotations

### PR DESCRIPTION
**Pull request by flaviostutz**
This is a continuation of #12 but as the code is very different, I created a new PR in order to cleanup things :)

Instead of using memoise, I used "lru_cache" from native functools (https://docs.python.org/3/library/functools.html), which does the same as memoise. Is it ok for you?

- PS: Sorry for the delay in handling this, but things got strange here too during those times :)

### CODE REVIEW

File: pyepsg_test.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+        #now it must be immediate as cache was already populated
+        start_time = datetime.datetime.now()
+        proj2 = pyepsg.get(21781)
+        e = datetime.datetime.now() - start_time
+        self.assertLess(e.total_seconds(), 0.01)
+        self.assertTrue('PROJCS["CH1903_LV03",GEOGCS["GCS_CH1903",DATUM["D_CH1903",SPHEROID["Bessel_1841",6377397.155,299.1528128]],PRIMEM["Greenwich",0],UNIT["Degree",0.017453292519943295]],PROJECTION["Hotine_Oblique_Mercator_Azimuth_Center"],PARAMETER["latitude_of_center",46.95240555555556],PARAMETER["longitude_of_center",7.439583333333333],PARAMETER["azimuth",90],PARAMETER["scale_factor",1],PARAMETER["false_easting",600000],PARAMETER["false_northing",200000],UNIT["Meter",1]]', proj2.as_esri_wkt())
```
Comment (rhattersley) [inline pyepsg_test.py:20]: Using the execution time to detect caching is not safe. For example, on a machine that accesses the internet through a caching web proxy even the first call to `pyepsg.get` may take less than 0.01 s. There are `cache_info()` and `cache_clear()` functions provided by the decorator to avoid the need for inferring behaviour in this way. But...

Effectively this is just testing the behaviour of functools.lru_cache which is not our responsibility.

So, thank you for adding a test, but I don't think the addition of the lru_cache decorator really warrants one. I suggest we remove this file for now.

File: requirements.txt [hunk 2]
```diff
@@ -1 +1,2 @@
 requests
+
```
Comment (rhattersley) [inline requirements.txt:2]: Ideally this spurious change would be excluded. (I'm guessing it's just left over from experimenting with `toolz` or similar.)

File: README.rst [hunk 3]
```diff
@@ -28,3 +28,8 @@ other forms::
 .. |build_status| image:: https://secure.travis-ci.org/rhattersley/pyepsg.png
    :alt: Build Status
    :target: https://travis-ci.org/rhattersley/pyepsg
+
+### Tests
+
+* In order to run unit tests, run `docker build .`. 
+* No specific Python dependencies are required to be installed on your machine.
```
Comment (rhattersley) [inline README.rst:35]: I'm not convinced that introducing Docker is appropriate. (Even if it were, it's not really related to the caching change in this PR, so it would be better in a separate PR. That way we can get the caching merged and discuss the use of Docker afterwards.)

Also, if we're not adding any tests then this (and the `Dockerfile`) is no longer needed.

**Review by rhattersley — CHANGES_REQUESTED** [review]
Thanks for the new PR. (y) And good idea to ditch Python 2 support and use the built-in decorator from Python 3. (y)

**Comment by flaviostutz** [comment]
Is there anything left so we can merge this PR? Maybe I lost something...

**Comment by rhattersley** [comment]
This PR is definitely a good improvement for the package, but before we can merge there are a few comments in my review that could do with being addressed. The main ones being the use of Docker, the test, and removing the remaining Python 2 bits.

**Comment by rhattersley** [comment]
Replaced by #17.
